pkg/config/lima_config_applier.go

```diff
@@ -188,3 +198,46 @@ func hasUserModeEmulationInstallationScript(limaCfg *limayaml.LimaYAML) (int, bo
 
 	return scriptIdx, hasCrossArchToolInstallationScript
 }
+
+func toggleSoci(limaCfg *limayaml.LimaYAML, enabled bool) {
+	idx, hasScript := hasSociInstallationScript(limaCfg)
```

Comment:
We may not need this. I remember it doesn't change the config persistently but from the default script every time.

Reply:
I think when I was testing this I saw the soci installation being appended to the override.yaml multiple times when I stop and started my VM. So, I thought the toggle was necessary. I'll try to add tests to double check if this is needed though.

Reply:
Update on this. The toggle isn't needed to stop the script from being appended multiple times, but I found that it is needed to remove the script and env var from the lima config when ```soci_snapshotter``` is set from true to false.